odoo/tools/monkeypatch.py

```diff
@@ -0,0 +1,223 @@
+
+    PyPDF2.generic.FloatObject = PatchedFloatObject
+    PyPDF2.filters.decompress = _decompress
+
+    return PyPDF2
+
+def monkeypatch_xmlrpc():
+    """
+        Fix invalid numbers like 0.000000000000-5684342
+        Because some PDF generators are building PDF streams with invalid numbers.
+    """
+
+    import re
+    from datetime import date, datetime
+    from odoo.loglevels import ustr
+    from odoo.tools.func import lazy
+    from odoo.tools.misc import frozendict, DEFAULT_SERVER_DATE_FORMAT, DEFAULT_SERVER_DATETIME_FORMAT
+
+    try:
+        import xmlrpc
+        import xmlrpc.client
+    except ImportError:
+        return None
+
+    # ustr decodes as utf-8 or latin1 so we can search for the ASCII bytes
+    # 	Char	   ::=   	#x9 | #xA | #xD | [#x20-#xD7FF]
+    XML_INVALID = re.compile(b'[\x00-\x08\x0B\x0C\x0F-\x1F]')
+
+    class OdooMarshaller(xmlrpc.client.Marshaller):
+        dispatch = dict(xmlrpc.client.Marshaller.dispatch)
+
+        def dump_frozen_dict(self, value, write):
+            value = dict(value)
+            self.dump_struct(value, write)
+        dispatch[frozendict] = dump_frozen_dict
+
+        # By default, in xmlrpc, bytes are converted to xmlrpclib.Binary object.
+        # Historically, odoo is sending binary as base64 string.
+        # In python 3, base64.b64{de,en}code() methods now works on bytes.
+        # Convert them to str to have a consistent behavior between python 2 and python 3.
+        def dump_bytes(self, value, write):
+            # XML 1.0 disallows control characters, check for them immediately to
+            # see if this is a "real" binary (rather than base64 or somesuch) and
+            # blank it out, otherwise they get embedded in the output and break
+            # client-side parsers
+            if XML_INVALID.search(value):
+                self.dump_unicode('', write)
+            else:
+                self.dump_unicode(ustr(value), write)
+        dispatch[bytes] = dump_bytes
+
+        def dump_datetime(self, value, write):
+            # override to marshall as a string for backwards compatibility
+            value = value.strftime(DEFAULT_SERVER_DATETIME_FORMAT) if value else False
+            self.dump_unicode(value, write)
+        dispatch[datetime] = dump_datetime
+
+        def dump_date(self, value, write):
+            value = value.strftime(DEFAULT_SERVER_DATE_FORMAT) if value else False
+            self.dump_unicode(value, write)
+        dispatch[date] = dump_date
+
+        def dump_lazy(self, value, write):
+            v = value._value
+            return self.dispatch[type(v)](self, v, write)
+        dispatch[lazy] = dump_lazy
+
+    # monkey-patch xmlrpc.client's marshaller
+    xmlrpc.client.Marshaller = OdooMarshaller
+    return xmlrpc
+
+def load_monkeypatched_module(name, monkeypatcher, parent_namespaces=None):
+    import sys
+    module = monkeypatcher()
+    if not module:
+        raise ModuleNotFoundError(name)
```

Comment:
*(No, it wasn't just about "ImportError" because there are other Exceptions being trapped )*
But on second thought, this is just to be removed and the module should just remain`None`
Do you agree?

![immagine](https://user-images.githubusercontent.com/1665365/186704327-3bed5ba7-b842-4641-a504-23ad497be811.png)

Reply:
I don't. It means that when importing the module in the business code, you will end up with `None`, which would not be expected

Reply:
I've done a change: if it's None, then I don't add it to the registry.
The old version actually did put odoo.misc.whatever.<module> = None btw